pandas/_libs/lib.pyi

```diff
@@ -12,7 +12,10 @@ from typing import (
 )
 
 import numpy as np
+from typing_extensions import TypeGuard
```

Comment:
`typing_extensions` isn't currently in `requirements-dev.txt`.

`TypeGuard` should probably be defined in `_typing.py` and imported from there, where we ensure it doesn't fail if `typing_extensions` isn't installed.

Reply:
good idea, moved